### PR TITLE
Fix ssl_lib functions for dtls 1.3

### PR DESCRIPTION
**Pull request by fwh-dc**

### CODE REVIEW

**Comment by fwh-dc** [comment]
CI failures are not relevant

File: ssl/ssl_lib.c [hunk 1]
```diff
@@ -2776,7 +2776,7 @@ int SSL_key_update(SSL *s, int updatetype)
     if (sc == NULL)
         return 0;
 
-    if (!SSL_CONNECTION_IS_TLS13(sc)) {
+    if (!(SSL_CONNECTION_IS_TLS13(sc) || SSL_CONNECTION_IS_DTLS13(sc))) {
```
Comment (t8m) [inline ssl/ssl_lib.c:2779]: Needs rebase to use the new macro.

**Review by t8m — CHANGES_REQUESTED** [review]

File: ssl/ssl_lib.c [hunk 2]
```diff
@@ -3786,7 +3787,7 @@ int SSL_export_keying_material_early(SSL *s, unsigned char *out, size_t olen,
     if (sc == NULL)
         return -1;
 
-    if (sc->version != TLS1_3_VERSION)
+    if (sc->version != TLS1_3_VERSION && sc->version != DTLS1_3_VERSION)
```
[the remark this answers is not in the document]
Comment (fwh-dc) [reply ssl/ssl_lib.c:3790]: I'm not sure actually it is a different check:

```
# define SSL_CONNECTION_IS_TLS13(s) (!SSL_CONNECTION_IS_DTLS(s) \
    && SSL_CONNECTION_GET_SSL(s)->method->version >= TLS1_3_VERSION \
    && SSL_CONNECTION_GET_SSL(s)->method->version != TLS_ANY_VERSION)
```

But I am not sure if will work anyways? I can give it a try.

[the remark this answers is not in the document]
Comment (fwh-dc) [reply ssl/ssl_lib.c:3790]: Otherwise I make the check nicer by checking on a `version1_3` variable.

[the remark this answers is not in the document]
Comment (t8m) [reply ssl/ssl_lib.c:3790]: Ah, you're right that checks the method version.

**Review by t8m — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the feature branch. Thank you for your contribution.

**Pull request closed without merging**